client/src/app/auth/auth.interceptor.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { AuthService } from './services/auth.service';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+
+  constructor(private authservice: AuthService) {}
+
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    let token = localStorage.getItem('token')
+    if (this.authservice.isLoggedIn() && token) {
```

Comment:
if `this.authservice.isLoggedIn ` is true it means that token exists, so you don't need to check it again

Reply:
I removed token from checking the token exists